sim/Dockerfile

```
@@ -1,18 +1,22 @@
-FROM ubuntu:22.04 AS builder
+FROM ubuntu:24.04 AS builder
 
 ARG TARGETARCH
 RUN echo "TARGETARCH : $TARGETARCH"
 
 RUN apt-get update && \
-  DEBIAN_FRONTEND=noninteractive apt-get install -y python3 build-essential cmake wget ninja-build
+  DEBIAN_FRONTEND=noninteractive apt-get install -y \
+  python3 build-essential cmake ninja-build libboost-dev libgsl-dev libxml2-dev \
+  libsqlite3-dev
 
-ENV NS_VERS 3.40
-RUN wget -O ns3.tar.bz2 https://www.nsnam.org/release/ns-allinone-$NS_VERS.tar.bz2
+ENV NS_VERS 3.41
+ADD --checksum=sha256:ce6a2c6a91d6979999da0a889cf0aa5db8798f387edf20672af9bcd966277aa9 \
+  https://www.nsnam.org/release/ns-allinone-$NS_VERS.tar.bz2 ns3.tar.bz2
 RUN tar xjf ns3.tar.bz2 && rm ns3.tar.bz2
 RUN mv /ns-allinone-$NS_VERS/ns-$NS_VERS /ns3
 
-ENV GO_VERS 1.21.5
-RUN wget -O go.tar.gz https://dl.google.com/go/go$GO_VERS.linux-$TARGETARCH.tar.gz
+ENV GO_VERS 1.22.3
+ADD --checksum=sha256:6c33e52a5b26e7aa021b94475587fce80043a727a54ceb0eee2f9fc160646434 \
```

I didn't realize Docker allowed verification by SHA256. How cool is that?!

Copy-paste error I assume:
```suggestion
ADD --checksum=sha256: 8920ea521bad8f6b7bc377b4824982e011c19af27df88a815e3586ea895f1b36 \
```

---

Oh wait, not a copy-paste error. This is the SHA256 for the arm64, which is why it fails on CI (amd64).

@larseggert How do we fix this?

---

Hm. Not sure if there is a way to do this. Do we actually need to install a custom go? 24.04 has go-1.22 via `apt`.

---

They used to have really outdated (multiple years) Go versions. If they have 1.22, we can use that.